server/schedule/replica_checker.go

```diff
@@ -197,9 +184,14 @@ func (r *ReplicaChecker) checkOfflinePeer(region *core.RegionInfo) *Operator {
 			return CreateRemovePeerOperator("removePendingOfflineReplica", r.cluster, OpReplica, region, peer.GetStoreId())
 		}
 
-		newPeer := r.SelectBestReplacedPeerToAddReplica(region, peer)
-		if newPeer == nil {
-			log.Debugf("[region %d] no best peer to add replica", region.GetId())
+		storeID, _ := r.SelectBestReplacementStore(region, peer)
+		if storeID == 0 {
+			log.Debugf("[region %d] no best store to add replica", region.GetId())
+			return nil
+		}
+		newPeer, err := r.cluster.AllocPeer(storeID)
+		if err != nil {
+			log.Error("alloc peer meet error:", err)
```

Comment:
I have a second thought here. It seems better to move the log into `AllocPeer` function.

Reply:
yeah, I check that found has already printed.